test/custom_verbose_configuration_tests.cpp

```diff
@@ -0,0 +1,74 @@
+#include "adobe/contract_checks.hpp"
+#include "portable_death_tests.hpp"
+#include <array>
+#include <cstdint>
+#include <cstdio>
+#include <exception>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <stdexcept>
+
+[[noreturn]] void adobe_contract_violated_verbose(const char *condition,
+  adobe::contract_violation_kind kind,
+  const char *file,
+  std::uint32_t line,
+  const char *message)
+{
+  std::array<char, 1024> output{};
+  (void)std::snprintf(output.data(),
+    sizeof(output),
+    "%s:%d: %s (%s). %s\n",
+    file,
+    static_cast<int>(line),
+    kind == adobe::contract_violation_kind::precondition ? "Precondition violated"
+                                                         : "Invariant not upheld",
+    condition,
+    message);
+  throw std::logic_error(output.data());
+}
+
+namespace {
+
+template<class F> void expect_throw(F f, const char *match)
+{
+  try {
+    f();
+  } catch (const std::exception &e) {
+    EXPECT_THAT(e.what(), testing::ContainsRegex(match));
+    return;
+  }
+}
+
+}// namespace
+
+#define STRINGIZE(x) STRINGIZE2(x)// NOLINT(cppcoreguidelines-macro-usage)
```

Comment:
Doc comments on these macros too please.  If this is a better idiom for capturing the line number we should use it in the other tests too, right?  *Is* it better?

Reply:
is it better?

Reply:
It's the same idiom you use in `verbose_configuration_tests.cpp` - it it better than what?

Reply:
I didn't recall having `LINE_STRING`, but since it's the same, we should DRY it, right?